### PR TITLE
qhull: update to 2020.2

### PR DESCRIPTION
**Pull request by olafmersmann**
Update qhull and revbump all reverse dependencies.

### CODE REVIEW

**Review by Piraty — CHANGES_REQUESTED** [review]
please squash the related changes into one qhull update commit (regarding `common/shlibs`: change the existing lines instead of appending to bottom)

**Comment by olafmersmann** [comment]
The build failures seem to stem from the changes in commit b8044ce629 and are unrelated to the changes in this merge request.

**Comment by ericonr** [comment]
Try adding `python3-pytest` to `checkdepends` for it.

**Comment by olafmersmann** [comment]
> Try adding `python3-pytest` to `checkdepends` for it.

I tried that in `master`:
```
❯ git diff
diff --git a/srcpkgs/python3-matplotlib/template b/srcpkgs/python3-matplotlib/template
index 3940b55d0f..c6acaf5861 100644
--- a/srcpkgs/python3-matplotlib/template
+++ b/srcpkgs/python3-matplotlib/template
@@ -9,6 +9,7 @@ hostmakedepends="pkg-config python3-setuptools python3-certifi python3-numpy"
 makedepends="python3-devel freetype-devel libpng-devel libqhull-devel"
 depends="python3-certifi python3-cycler python3-kiwisolver
  python3-numpy python3-Pillow python3-parsing python3-dateutil"
+checkdepends="$depends python3-pytest>=3.6"
 short_desc="Python3 2D/3D plotting library"
 maintainer="Alessio Sergi <al3hex@gmail.com>"
 license="custom:matplotlib, BSD-3-Clause, MIT"
```
but still get
```
❯ ./xbps-src check -f python3-matplotlib
*snip*
lib/matplotlib/__init__.py:174: in <module>
    _check_versions()
lib/matplotlib/__init__.py:159: in _check_versions
    from . import ft2font
E   ImportError: cannot import name 'ft2font' from partially initialized module 'matplotlib' (most likely due to a circular import) (/builddir
/matplotlib-3.3.3/lib/matplotlib/__init__.py)
=========================== short test summary info ============================
ERROR  - ImportError: cannot import name 'ft2font' from partially initialized...
!!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!
=============================== 1 error in 0.30s ===============================
=> ERROR: python3-matplotlib-3.3.3_1: do_check: 'python3 -m pytest ${make_check_args} ${make_check_target}' exited with 2
=> ERROR:   in do_check() at common/build-style/python3-module.sh:28
```

I'll investigate further. If I find a solution, it would go into a separate merge request, right?

**Comment by ericonr** [comment]
Not necessary, it can go in this same PR. Specifying a pytest version isn't necessary (and is actually wrong), it will always use the one specified in `srcpkgs/`.

Try cleaning the directories and running `./xbps-src -Q pkg python3-matplotlib` to be sure.

File: srcpkgs/qhull/template [hunk 1]
```diff
@@ -1,16 +1,21 @@
...
 pkgname=qhull
-version=2019.1
+version=2020.2
 revision=1
 build_style=cmake
+make_build_target="all libqhull"
 short_desc="Computes volumes, surface areas and approximations to the convex hull"
 maintainer="Martin Riese <grauehaare@gmx.de>"
-license="BSD"
+license="Qhull"
```
Comment (ahesford) [inline srcpkgs/qhull/template:9]: You will need to `vlicense` a copy of this license in `post_install`, as per the terms of the license.

File: srcpkgs/qhull/template [hunk 2]
```diff
@@ -1,16 +1,21 @@
...
 homepage="http://www.qhull.org"
+changelog="http://www.qhull.org/src/Changes.txt"
 distfiles="https://github.com/qhull/qhull/archive/$version.tar.gz"
-checksum=cf7235b76244595a86b9407b906e3259502b744528318f2178155e5899d6cf9f
+checksum=59356b229b768e6e2b09a701448bfa222c37b797a84f87f864f97462d8dbc7c5
 
 post_install() {
+	# Manually install libqhull (deprecated by the package author)
+	cp -r ${wrksrc}/build/libqhull.so* ${DESTDIR}/usr/lib
```
Comment (ahesford) [inline srcpkgs/qhull/template:17]: Have you tried dropping the `libqhull` target and removing `${FILESDIR}/qhull.pc` to see if the dependants will link against the non-deprecated `libqhull_r`? If everything we ship links against the new library, we should drop the deprecated, non-default targets.

File: srcpkgs/qhull/template [hunk 3]
```diff
@@ -1,16 +1,21 @@
...
+
 	install -Dm644 ${FILESDIR}/qhull.pc ${DESTDIR}/usr/lib/pkgconfig/qhull.pc
 	sed -i "s|@VERSION@|${version}|" ${DESTDIR}/usr/lib/pkgconfig/qhull.pc
 	# convenience links for backwards compatibility
```
Comment (ahesford) [inline srcpkgs/qhull/template:21]: Can you confirm that the "compatibility links" in in `/usr/include` are necessary to build the dependants? If not, they should be dropped.

**Review by ahesford — CHANGES_REQUESTED** [review]

File: srcpkgs/qhull/template (hunk 1 above)
Comment (olafmersmann) [reply srcpkgs/qhull/template:9]: Done.

File: srcpkgs/qhull/template (hunk 2 above)
Comment (olafmersmann) [reply srcpkgs/qhull/template:20]: Yes. At least `octave` and `python3-matplotlib` still require `libqhull`.

File: srcpkgs/qhull/template (hunk 3 above)
Comment (olafmersmann) [reply srcpkgs/qhull/template]: Yes, required for `octave`.

**Comment by ahesford** [comment]
I have rolled this into some other updates and will merge soon. The `python3-matplotlib` tests are problematic for several reasons:
- The `PYTHONPATH` would need to be set to find compiled extensions, which is the source of the failure here.
- If you set `PYTHONPATH` to include `build/lib*` where the compiled extensions are, the `matplotlib` module does not include the baseline images needed for the tests.
- Copying the shared objects from `build/lib*` to the corresponding directories of `lib` and setting `PYTHONPATH=lib` allows the tests to run.

Unfortunately, `matplotlib` tests are evaluated by comparing output images to baseline images produced upstream, which is incredibly fragile because of differences in fonts and layout. Just disabling the tests is the right thing to do here.

Of course, if you're disabling tests, there is no value in pulling in `checkdepends`, so I've dropped that change.

**Pull request closed without merging**